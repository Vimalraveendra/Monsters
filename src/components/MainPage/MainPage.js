import React, { Component } from "react";
import CardList from "../CardList/CardList";
import "./MainPage.css";
import SearchBox from "../SearchBox/SearchBox";

class MainPage extends Component {
  componentDidMount() {
    this.props.fetchCall(fetch);
  }
  filteredMonsters = () => {
    return this.props.monsters.filter(monster => {
      return monster.name
        .toLowerCase()
        .includes(this.props.searchField.toLocaleLowerCase());
    });
  };
  render() {
    return (
      <div className="App">
        <h1>Monsters Rolodex</h1>
        <SearchBox
          placeholder="search monsters"
          handleChange={this.props.handleChange}
        />
        <CardList monsters={this.filteredMonsters()} />
      </div>
    );
  }
}
export default MainPage;
