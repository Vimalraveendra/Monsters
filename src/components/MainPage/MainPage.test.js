import { shallow } from "enzyme";
import React from "react";
import MainPage from "./MainPage";

let wrapper;
beforeEach(() => {
  const mockProps = {
    fetchCall: jest.fn(),
    monsters: [],
    searchField: ""
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

it("should  render the Mainpage without crashing", () => {
  expect(wrapper).toMatchSnapshot();
});

describe("filterMonsters", () => {
  it("should filter the monsters correctly", () => {
    expect(wrapper.instance().filteredMonsters()).toEqual([]);
  });

  it("should filter the monsters with searchField item present", () => {
    const mockProps = {
      fetchCall: jest.fn(),
      monsters: [
        {
          id: 123,
          name: "John snow",
          email: "Johnsnow@gmail.com"
        }
      ],
      searchField: "Jo"
    };
    wrapper = shallow(<MainPage {...mockProps} />);
    expect(wrapper.instance().filteredMonsters()).toEqual([
      {
        id: 123,
        name: "John snow",
        email: "Johnsnow@gmail.com"
      }
    ]);
  });

  it("should filter the monsters with searchField item present", () => {
    const mockProps = {
      fetchCall: jest.fn(),
      monsters: [
        {
          id: 123,
          name: "John snow",
          email: "Johnsnow@gmail.com"
        }
      ],
      searchField: "hel"
    };
    wrapper = shallow(<MainPage {...mockProps} />);
    expect(wrapper.instance().filteredMonsters()).toEqual([]);
  });
});
