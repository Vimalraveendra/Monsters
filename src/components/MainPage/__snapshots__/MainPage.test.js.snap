// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should  render the Mainpage without crashing 1`] = `ShallowWrapper {}`;
